src/js/components/Marathon.jsx

```diff
@@ -44,21 +46,24 @@ var Marathon = React.createClass({
   },
 
   componentDidMount: function () {
-    this.onRouteChange();
+    PluginActions.requestPlugins();
 
+    this.onRouteChange();
     this.bindKeyboardShortcuts();
 
-    PluginStore.once(PluginEvents.BOOTSTRAP_COMPLETE, () => {
-      this.startPolling();
-    });
-
-    PluginStore.bootstrap();
+    PluginStore.on(PluginEvents.CHANGE, this.onPluginStoreChange);
   },
 
   componentWillReceiveProps: function () {
     this.onRouteChange();
   },
 
+  onPluginStoreChange: function () {
+    if (PluginStore.getPluginLoadingState() > States.STATE_LOADING) {
```

Comment:
I've removed the event, as bootstrapping is part of the plugin loading process. I thought the whole idea of the States const was to enable such conditions, but you're right they're kinda brittle. I'll refactor the condition.  *BTW: There is already an error state*